pennylane/transforms/qcut.py

```diff
@@ -210,3 +213,121 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
+
+    This operation is differentiable. The ``prepare_nodes`` and ``measure_nodes`` arguments are both
+    sequences of size ``len(communication_graph.nodes)`` that describe the order of indices in the
+    ``tensors`` with respect to to the :class:`~.PrepareNode` and :class`~.MeasureNode` edges in the
+    communication graph.
+
+    Args:
+        tensors (Sequence): the tensors to be contracted
+        communication_graph (MultiDiGraph): the communication graph determining connectivity between
+            the tensors
+        prepare_nodes (Sequence[PrepareNode]): a sequence of size ``len(communication_graph.nodes)``
+            that determines the order of preparation indices in each tensor
+        measure_nodes (Sequence[MeasureNode]): a sequence of size ``len(communication_graph.nodes)``
+            that determines the order of measurement indices in each tensor
+        use_opt_einsum (bool): Determines whether to use the
+            [opt_einsum](https://dgasmith.github.io/opt_einsum/) package. This package is useful for
+            tensor contractions of large networks but must be installed separately using, e.g.,
+            ``pip install opt_einsum``.
+
+    Returns:
+        float or array-like: the result of contracting the tensor network
+
+    **Example**
+
+    .. note::
+
+        This function is designed for use as part of the circuit cutting workflow. Check out the
+        :doc:`transforms </code/qml_transforms>` page for more details.
+
+    We first set up the tensors and their corresponding :class:`~.PrepareNode` and
+    :class`~.MeasureNode` orderings:
+
+    .. code-block:: python
+
+        t = [np.arange(4), np.arange(4, 8)]
+        p = [[], [qcut.PrepareNode(wires=0)]]
+        m = [[qcut.MeasureNode(wires=0)], []]
+
+    The communication graph describing edges in the tensor network must also be constructed:
+
+    .. code-block:: python
+
+        g = MultiDiGraph([(0, 1, {"pair": (m[0][0], p[1][0])})])
+
+    The network can then be contracted using:
+
+    >>> qcut.contract_tensors(t, g, p, m)
+    38
+    """
+    # pylint: disable=import-outside-toplevel
+    if use_opt_einsum:
+        try:
+            from opt_einsum import contract, get_symbol
+        except ImportError as e:
+            raise ImportError(
+                "The opt_einsum package is required when use_opt_einsum is set to "
+                "True in the contract_tensors function. This package can be "
+                "installed using:\npip install opt_einsum"
+            ) from e
+    else:
+        from string import ascii_letters as symbols
+
+        from pennylane.math import einsum as contract
+
+        def get_symbol(i):
+            if i >= len(symbols):
+                raise ValueError(
+                    "Set the use_opt_einsum argument to True when applying more than "
+                    f"{len(symbols)} wire cuts to a circuit"
+                )
+            return symbols[i]
```

Comment:
These could be defined at the module level, right? And then here you could just have

```python
else:
    contract = qml.masth.einsum
    get_symbol = get_symbol_no_einsum

Reply:
Done: f537e28788d5fadfd5166ca38279fd8c9594261a